lsp/source/server.mts

```diff
@@ -373,27 +373,46 @@ documents.onDidOpen(async ({ document }) => {
 })
 
 // Buffer up changes to documents so we don't stack transpilations and become unresponsive
-const changeQueue = new Set<TextDocument>()
-setInterval(() => {
-  if (changeQueue.size === 0) return
-  Array.from(changeQueue).forEach((document) => {
-    updateDiagnostics(document)
-  })
-  changeQueue.clear()
-}, 50)
+let changeQueue = new Set<TextDocument>()
+let executeTimeout: Promise<void> | undefined
+async function executeQueue() {
+  // Cancel updating any other documents while running queue of primary changes
+  if (runningDiagnosticsUpdate) {
+    runningDiagnosticsUpdate.isCanceled = true
+  }
+  // Reset queue to allow accumulating jobs while this queue runs
+  const changed = changeQueue
+  changeQueue = new Set
+  console.log("executeQueue", changed.size)
+  // Run all jobs in queue (preventing livelock).
+  for (const document of changed) {
+    await updateDiagnosticsForDoc(document)
+  }
+  // Allow executeQueue() again, and run again if there are new jobs now.
+  // Otherwise, schedule update of all other documents.
+  executeTimeout = undefined
+  if (changeQueue.size) {
+    scheduleExecuteQueue()
+  } else {
+    scheduleUpdateDiagnostics(changed)
+  }
+}
+async function scheduleExecuteQueue() {
+  // Schedule executeQueue() if there isn't one already running or scheduled
+  if (executeTimeout) return
+  if (!changeQueue.size) return
+  await (executeTimeout = setTimeout(50))
```

Comment:
Let's change this to `25` or `16` since it is the minimum time before processing rather than the maximum.